tooling/api/src/tauri.ts

```diff
@@ -55,6 +55,29 @@ function transformCallback(
   return identifier
 }
 
+class Channel {
```

Comment:
I would refactor this to be more similar to the `Websocket` or `BroadcastChannel` api i.e. the constructor takes no arguments and you can listen to events by overriding the `on message` property or calling `.addEventListener('message', () => {})`